visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts

```diff
@@ -32,19 +21,23 @@ export class FileChooserController {
 			for (const file of element.files) {
 				const isCompressed = file.name.endsWith(".gz")
 				const reader = new FileReader()
-				isCompressed ? reader.readAsArrayBuffer(file) : reader.readAsText(file, "UTF-8")
+				if (isCompressed) {
+					reader.readAsArrayBuffer(file)
+				} else {
+					reader.readAsText(file, "UTF-8")
+				}
 
-				reader.onloadstart = () => {
+				reader.addEventListener("loadstart", () => {
 					this.storeService.dispatch(setIsLoadingFile(true))
-				}
+				})
 
-				reader.onload = event => {
+				reader.addEventListener("load", event => {
```

Comment:
Why don't we use the existing event listeners? Creating new ones and NOT removing them is a bad idea.

Reply:
Good catch. This was actually auto fixed. I'll deactivate the rule for now.